tests/Controller/ShippingMethodsNewApiTest.php

```diff
@@ -0,0 +1,311 @@
+    {
+        $data =
+<<<EOT
+        {
+            "name": "InPost"
+        }
+EOT;
+
+        $this->client->request(
+            'PUT',
+            $this->getShippingMethodUrl($this->loadFixturesAndGetShippingMethod('ups')),
+            [],
+            [],
+            self::$authorizedHeaderWithDenied,
+            $data
+        );
+
+        $this->assertResponse(
+            $this->client->getResponse(),
+            'api/authentication/access_denied_response',
+            Response::HTTP_UNAUTHORIZED
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function it_accepts_to_put_shipping_method_for_authenticated_users(): void
+    {
+        $token = $this->logInAsAdminAndGetToken();
+
+        $data =
+<<<EOT
+        {
+            "name": "InPost"
+        }
+EOT;
+
+        $this->client->request(
+            'PUT',
+            $this->getShippingMethodUrl($this->loadFixturesAndGetShippingMethod('ups')),
+            [],
+            [],
+            $this->getAuthorizedHeaderWithAccept($token),
+            $data
+        );
+
+        $this->assertResponseCode($this->client->getResponse(), Response::HTTP_OK);
+    }
+
+    /**
+     * @test
+     */
+    public function it_denies_to_delete_shipping_method_for_not_authenticated_users(): void
+    {
+        $this->client->request(
+            'DELETE',
+            $this->getShippingMethodUrl($this->loadFixturesAndGetShippingMethod('ups')),
+            [],
+            [],
+            self::$authorizedHeaderWithDenied
+        );
+
+        $this->assertResponse(
+            $this->client->getResponse(),
+            'api/authentication/access_denied_response',
+            Response::HTTP_UNAUTHORIZED
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function it_accepts_to_delete_shipping_method_for_authenticated_users(): void
+    {
+        $token = $this->logInAsAdminAndGetToken();
+
+        $this->client->request(
+            'DELETE',
+            $this->getShippingMethodUrl($this->loadFixturesAndGetShippingMethod('ups')),
+            [],
+            [],
+            $this->getAuthorizedHeaderWithAccept($token)
+        );
+
+        $this->assertResponseCode($this->client->getResponse(), Response::HTTP_NO_CONTENT);
+    }
+
+    private function getAuthorizedHeaderWithAccept(string $token): array
+    {
+        return [
+            'HTTP_Authorization' => 'Bearer ' . $token,
+            'HTTP_ACCEPT' => 'application/ld+json',
+            'CONTENT_TYPE' => 'application/ld+json',
+        ];
+    }
+
+    private function getShippingMethodUrl(?ShippingMethodInterface $shippingMethod = null): string
+    {
+        $url = '/new-api/shipping-methods';
+
+        return $shippingMethod !== null ? sprintf('%s/%s', $url, $shippingMethod->getCode()) : $url;
+    }
+
+    private function loadFixturesAndGetShippingMethod(string $shippingMethodName): ShippingMethodInterface
```

Comment:
this method and next one do more than one thing, IMO they aren't the true SOLID method, maybe two separately methods will be better?

Reply:
I think that in this simple tests, Joining this two methods reduces the amount of redundant lines. Maybe others give opinion? @lchrusciel @GSadee  ?

Reply:
From the perspective of the method alone I think it would be ok to leave it as it is. However taking into account line https://github.com/Sylius/Sylius/pull/11497/files#diff-f11056c5b09a5942258aa228edeb658dR56 or https://github.com/Sylius/Sylius/pull/11497/files#diff-f11056c5b09a5942258aa228edeb658dR145 I would say, that separation of concerns may be a good idea. So in the first method we should load fixtures data (which is covered by ApiTestCase, so maybe new method is not needed at all), and the second should fetch the shipping method from database.